### PR TITLE
fix(curriculum): revise some of the tests in lab-travel-agency-page

### LINKED ISSUE #56861: [ Full Stack Cert ] - update tests for travel page
There are a few updates we need to make for the travel page lab. 

First, remove all jQuery 

Update tests to use specific assert methods where applicable 

Also, this hint does not check for any content between the paragraph tags

```md
You should have a `p` element introducing the travel opportunities below the `h1` element.
```

This currently passes

```html
<body>
  <h1>travel title</h1>
  <p></p>
</body>
```

It would be better to split this hint up into two tests.

One test to check if the paragraph element is below the `h1`
One test to check if the paragraph element actually has some sort of content. 

Same issue here. This only checks if there is a paragraph element below the `h2`. It doesn’t check for any content between the tags.

```md
You should have a `p` element introducing briefly the various packages below your first `h2` element.
```

```js
assert($('p')[1]?.previousElementSibling.tagName === 'H2');
```

This currently passes 

```html
<body>
  <h1>travel title</h1>
  <p>something</p>
  <h2>Packages</h2>
  <p></p>
</body>
```

It would be better to split this hint up into two tests.

One test to check if the paragraph element is below the `h2`
One test to check if the paragraph element actually has some sort of content. 

Here is the file you need to update 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md

### PR DESCRIPTION
**Pull request by CracktheDom**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56861

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 1]
```diff
@@ -90,7 +90,13 @@ assert.lengthOf(document.querySelectorAll('h1'), 1)
 You should have a `p` element introducing the travel opportunities below the `h1` element.
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:90]:
```suggestion
You should have a `p` element below the `h1` element.
```

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 2]
```diff
@@ -90,7 +90,13 @@ assert.lengthOf(document.querySelectorAll('h1'), 1)
...
 
 ```js
-assert($('p')[0]?.previousElementSibling.tagName === 'H1');
+// Checks that a <p> tag follows a <h1> tag
+const h1Element = document.querySelector('h1');
+const pElement = document.querySelector('p');
+assert(h1Element && pElement && h1Element.nextElementSibling === pElement)
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:96]:
```suggestion
assert.strictEqual(h1Element.nextElementSibling, pElement)
```

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 3]
```diff
@@ -90,7 +90,13 @@ assert.lengthOf(document.querySelectorAll('h1'), 1)
...
+
+// Check that <p> element contains text
+assert(pElement && pElement.textContext.trim() !== "")
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:99]:
```suggestion
assert.isNotEmpty(pElement.textContext.trim())
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
The issue requested breaking the hint/test into two, one for the order of the elements, one for the content of the `p` element.

Also please use [specific assert methods](https://www.chaijs.com/api/assert/), and [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md (hunk 2 above)
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:96]:
````suggestion
assert.strictEqual(document.querySelector('h1')?.nextElementSibling, document.querySelector('p'))
```
````
Splitting into two tests.

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md (hunk 3 above)
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:99]:
````suggestion
Your first `p` element should introduce the travel opportunities.

```js
assert.isNotEmpty(document.querySelector('p')?.textContent?.trim());
````

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 4]
```diff
@@ -102,13 +108,19 @@ assert.equal(document.querySelectorAll('h2')[0]?.innerText, 'Packages');
 You should have a `p` element introducing briefly the various packages below your first `h2` element.
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:108]:
```suggestion
You should have a `p` element below your first `h2` element.
```

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 5]
```diff
@@ -102,13 +108,19 @@ assert.equal(document.querySelectorAll('h2')[0]?.innerText, 'Packages');
...
 
 ```js
-assert($('p')[1]?.previousElementSibling.tagName === 'H2');
+// Check that a <p> tag directly follows a <h2> tag
+const h2Element = document.querySelectorAll('h2')[0];
+const pElement = document.querySelector('p');
+assert(h2Element && pElement && h2Element.nextElementSibling === pElement)
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:114]:
````suggestion
assert.strictEqual(document.querySelector('h2')?.nextElementSibling, document.querySelectorAll('p')?.[1])
```
````
Splitting into two tests.

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md [hunk 6]
```diff
@@ -102,13 +108,19 @@ assert.equal(document.querySelectorAll('h2')[0]?.innerText, 'Packages');
...
+
+// Check that <p> element contains text
+assert(pElement && pElement.textContext.trim() !== "")
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:117]:
````suggestion
Your second `p` element should introduce briefly the various packages.

```js
assert.isNotEmpty(document.querySelectorAll('p')?.[1]?.textContent?.trim())
````

**Review by CracktheDom — COMMENTED** [review]
Add suggestions

**Review by majestic-owl448 — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md (hunk 1 above)
Comment (gikf) [reply curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:90]: Looks like two of these were missed missed

File: curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md (hunk 4 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-travel-agency-page/669e2f60e83c011754f711f9.md:108]: You have missed this one @CracktheDom

**Review by gikf — APPROVED** [review]

**Pull request merged**